platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JavaCompilerArgumentsBuilder.java

```diff
@@ -210,10 +210,9 @@ private void addMainOptions(List<String> compilerArgs) {
             List<File> annotationProcessorPath = spec.getAnnotationProcessorPath();
             if (annotationProcessorPath == null || annotationProcessorPath.isEmpty()) {
                 args.add("-proc:none");
-            } else {
-                args.add("-processorpath");
-                args.add(Joiner.on(File.pathSeparator).join(annotationProcessorPath));
             }
+            args.add("-processorpath");
+            args.add(Joiner.on(File.pathSeparator).join(annotationProcessorPath));
```

Comment:
You need to coerce `null` to an empty iterable if you want to move this out of the `else` branch.

Reply:
Right, sorry for the noise.This was just a quick hack to see whether it would have any effect on the SantTracker smoke tests and it didn't. I need to look closer into how annotation processing is being done in that particular build, I'll move the PR back to draft state.